pkg/configurator/crd_client.go

```diff
@@ -0,0 +1,183 @@
+package configurator
+
+import (
+	"strings"
+
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/openservicemesh/osm/pkg/announcements"
+	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha1"
+	"github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned"
+	informers "github.com/openservicemesh/osm/pkg/gen/client/config/informers/externalversions"
+	k8s "github.com/openservicemesh/osm/pkg/kubernetes"
+	"github.com/openservicemesh/osm/pkg/kubernetes/events"
+)
+
+const (
+	meshConfigInformerName = "MeshConfig"
+	meshConfigProviderName = "OSM"
+)
+
+func newConfiguratorWithCRDClient(meshConfigClientSet versioned.Interface, stop <-chan struct{}, osmNamespace string) *CRDClient {
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(
+		meshConfigClientSet,
+		k8s.DefaultKubeEventResyncInterval,
+		informers.WithNamespace(osmNamespace),
+	)
+	informer := informerFactory.Config().V1alpha1().MeshConfigs().Informer()
+	crdClient := CRDClient{
+		informer:     informer,
+		cache:        informer.GetStore(),
+		cacheSynced:  make(chan interface{}),
+		osmNamespace: osmNamespace,
+	}
+
+	// configure listener
+	eventTypes := k8s.EventTypes{
+		Add:    announcements.MeshConfigAdded,
+		Update: announcements.MeshConfigUpdated,
+		Delete: announcements.MeshConfigDeleted,
+	}
+	informer.AddEventHandler(k8s.GetKubernetesEventHandlers(meshConfigInformerName, meshConfigProviderName, nil, eventTypes))
+
+	// start listener
+	crdClient.startMeshConfigListener(stop)
```

Comment:
Would it make sense to use `go crdClient.startMeshConfigListener(stop)` here and remove the `go func() {` below?

Reply:
I wonder if there is a way to distinguish which golang method should be called using goroutine. I don't find a naming convention for this purpose in the community.

I think both approaches are fine here. If we use a goroutine here, I would like to change the function name to `runMeshConfigListener()`. So we can define our own convention that, the `runXYZ()` functions are goroutines.